renpy/display/transform.py

```diff
@@ -43,12 +45,127 @@ class Camera(renpy.object.Object):
+        self.relative: float = relative
+
+    @classmethod
+    def from_any(cls, other):
+        if isinstance(other, cls):
+            return other
+
+        elif type(other) is float:
+            return cls(other, other)
+
+        else:
+            raise TypeError(f"Cannot convert {other.__class__} to DualAngle")
+
+    def __add__(self, value: "DualAngle") -> "DualAngle":
+        if isinstance(value, DualAngle):
+            return DualAngle(self.absolute + value.absolute, self.relative + value.relative)
+
+        return NotImplemented
+
+    def __sub__(self, value: "DualAngle") -> "DualAngle":
+        if isinstance(value, DualAngle):
+            return DualAngle(self.absolute - value.absolute, self.relative - value.relative)
+
+        return NotImplemented
+
+    def __mul__(self, value: int | float) -> "DualAngle":
+        if isinstance(value, (int, float)):
+            return DualAngle(self.absolute * value, self.relative * value)
+
+        return NotImplemented
+
+    __rmul__ = __mul__
+
+    # Those methods are used in TransformState to help polar properties calculations.
+    @staticmethod
+    def get_pos_polar_vector(ts: "TransformState") -> tuple[float, float]:
+        """
+        Return a tuple of (x, y) representing the position of the anchor point.
+        """
+
+        xpos = position(first_not_none(ts.xpos, ts.inherited_xpos, 0))
+        xpos = absolute.compute_raw(xpos, ts.available_width)
+
+        ypos = position(first_not_none(ts.ypos, ts.inherited_ypos, 0))
+        ypos = absolute.compute_raw(ypos, ts.available_height)
+
+        xaround = absolute.compute_raw(position(ts.xaround), ts.available_width)
+        yaround = absolute.compute_raw(position(ts.yaround), ts.available_height)
+
+        return (xpos - xaround, ypos - yaround)
+
+    @staticmethod
+    def get_anchor_polar_vector(ts: "TransformState") -> tuple[tuple[float, float], tuple[float, float]]:
+        """
+        Returns a 2-tuple of 2-tuples,
+        where the first small tuple is absolute and the second tuple is relative,
+        and the first element of each tuple is in x and the second in y.
+        They represent the vector from the anchoraround point to the final anchor point.
+        """
+
+        xanchoraround = position(ts.xanchoraround)
+        yanchoraround = position(ts.yanchoraround)
+        xanchor = position(first_not_none(ts.xanchor, ts.inherited_xanchor, 0))
+        yanchor = position(first_not_none(ts.yanchor, ts.inherited_yanchor, 0))
+
+        absolute_vector = (
+            xanchor.absolute - xanchoraround.absolute,
+            yanchor.absolute - yanchoraround.absolute,
+        )
+        relative_vector = (
+            xanchor.relative - xanchoraround.relative,
+            yanchor.relative - yanchoraround.relative,
+        )
+
+        return absolute_vector, relative_vector
+
+    @staticmethod
+    def set_pos_from_angle_and_radius(ts: "TransformState", angle: float, radius: float):
```

Comment:
I'm a bit skeptical about having a static setter method, and not on the type it sets on. I think I get the idea of moving the get methods out of the massive TransformState object, but that one seems weird to me.